### ISSUE TITLE
Delay after last command while shrinking

### ISSUE THREAD

**Issue by alexander-zibert** [dubzzz/fast-check#462]
## 💬 Question and Help

One property I want to check while running a model-based UI test is that no errors were thrown in the app (e.g. with `window.onerror` and a `console.error` mock). This check mostly works fine, but not while shrinking. Often the errors are only detected after the asyncModelRun. This is due to the async nature of the occurrence and communication (e.g. with puppeteer) of the errors, which leads to false-positive test results and prevents optimal shrinking.

Are there possibilities in fast-check to do a post-condition check after an `asyncModelRun` has finished successfully?

My current solution is adding the delay (200 ms) after every command. But this considerably slows down the test.

What I already tried:

* I already tried `asyncProperty` with the `afterEach` hook, but throwing there has no influence on shrinking and only fails the whole test.
* I also considered always adding a `CheckErrorCommand` at the end of the command list, which checks that the number of errors is zero after a specific delay of like 200 ms. Still, I didn't get it to work, and this approach seems to ignore the advice never to change the inputs of the predicate function.

**Comment by dubzzz** [comment]
Hello, if I'm not wrong, the following snippet should do what you're looking for:

```js
test('here is a command-based test', async () => {
  await fc.assert(
    fc.asyncProperty(fc.commands(allCommands, 100), cmds => {
      const s = () => ({ model: /*model*/, real: /*real*/ });
      await fc.asyncModelRun(s, cmds);
      await sleep(200);     // TODO replace with your sleep function
      await checkConsole(); // TODO replace with your checkConsole function
    })
  );
});
```

Please let me know if it solves your issue ;)

**Comment by alexander-zibert** [comment]
It works like a charm! I'm still trying to wrap my head around model-based testing. It seems that something was wrong with my code when I tried the same code. Thank you very much for creating this great tool.
